pkg/controllers/v1alpha1/dataload/implement.go

```diff
@@ -325,12 +344,38 @@ func (r *DataLoadReconcilerImplement) reconcileExecutingDataLoad(ctx reconcileRe
 	// 2. install the helm chart if not exists and requeue
 	if !existed {
 		log.Info("DataLoad job helm chart not installed yet, will install")
-		valueFileName, err := r.generateDataLoadValueFile(ctx.DataLoad)
+
+		targetDataset, err := utils.GetDataset(r.Client, ctx.DataLoad.Spec.Dataset.Name, ctx.DataLoad.Spec.Dataset.Namespace)
 		if err != nil {
-			log.Error(err, "failed to generate dataload chart's value file")
-			return utils.RequeueIfError(err)
+			log.Error(err, "targetDataset not exists", "targetDataset", releaseName, "namespace", ctx.DataLoad.Spec.Dataset.Name)
+		}
+
+		_, boundedRuntime := utils.GetRuntimeByCategory(targetDataset.Status.Runtimes, common.AccelerateCategory)
+
+		chartName := ""
+		valueFileName := ""
+		switch boundedRuntime.Type {
+		case common.ALLUXIO_RUNTIME:
+			valueFileName, err = r.generateDataLoadValueFile(ctx.DataLoad)
+			if err != nil {
+				log.Error(err, "failed to generate dataload chart's value file")
+				return utils.RequeueIfError(err)
+			}
+			chartName = utils.GetChartsDirectory() + "/" + cdataload.DATALOAD_CHART + "/" + common.ALLUXIO_RUNTIME
```

Comment:
how about make it a function which return chartName according to runtimeName?

Reply:
thx, will refeactor next PR